tsdb/engine/tsm1/compact.go

```diff
@@ -417,11 +426,18 @@ func (c *DefaultPlanner) PlanOptimize() ([]CompactionGroup, int64) {
 		cGroups = append(cGroups, cGroup)
 	}
 
+	// The third return value is the generation length.
```

Comment:
The fact that the third return valued is the generation length should be documented in the function header comment, or in the interface it is implementing.